analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/CallModelStateIsValid.cs

```diff
@@ -90,6 +87,11 @@ private static void ProcessCodeBlock(SonarCodeBlockStartAnalysisContext<SyntaxKi
         }
     }
 
+    private static bool HasApiControllerAttribute(ITypeSymbol type) =>
+        type is not null
+        && (type.HasAttribute(KnownType.Microsoft_AspNetCore_Mvc_ApiControllerAttribute)
+            || HasApiControllerAttribute(type.BaseType));
```

Comment:
To avoid the recursion and make it a bit more readable, you can use the inherited version of the attribute check:

```csharp
    private static bool HasApiControllerAttribute(ITypeSymbol type) =>
        type.GetAttributesWithInherited().Any(x => x.AttributeClass.DerivesOrImplements(KnownType.Microsoft_AspNetCore_Mvc_ApiControllerAttribute));
```